Source/Plugins/Core/com.equella.serverbase/src/com/tle/core/hibernate/equella/service/impl/InitialiserServiceImpl.java

```
@@ -99,7 +102,27 @@ public void entitySimplified(Object old, Object newObj) {
    */
   private void evictFromSession(Set<Object> os) {
     for (Object o : os) {
-      getHibernateTemplate().evict(o);
+      try {
+        getHibernateTemplate().evict(o);
+      } catch (IllegalArgumentException iae) {
+        if (iae.getMessage().startsWith("Non-entity object instance passed to evict")) {
+          // This is being thrown in DefaultEvictEventListener.java:94
+          // Does not appear to be a problem in this case, so trapping
+          // and ignoring this specific case
+          LOGGER.debug(
+              "Ignoring error with type ["
```

would it be possible/make sense to filter the objects for valid ones?
Or is swallowing the error the only/most effective way to handle this?

---

Please see response above.